src/modules/sub-items/components/sub-items-list/instant.filter.component.js

```diff
@@ -0,0 +1,93 @@
+import React, { useEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+
+const FILTER_TIMEOUT = 200;
+
+const InstantFilter = (props) => {
+    const _refInstantFilter = useRef(null);
+    const [filterQuery, setFilterQuery] = useState('');
+    const [itemsMap, setItemsMap] = useState([]);
+    const [filterTimeout, setFilterTimeout] = useState('null');
+
+    useEffect(() => {
+        const items = [..._refInstantFilter.current.querySelectorAll(`.${props.itemClass}`)];
+        const itemsMap = items.reduce(
+            (total, item) => [
+                ...total,
+                {
+                    label: item.textContent.toLowerCase(),
+                    element: item,
+                },
+            ],
+            []
+        );
+
+        setItemsMap(itemsMap);
+    }, []);
+
+    useEffect(() => {
+        window.clearTimeout(filterTimeout);
+
+        const filterActionTimeout = window.setTimeout(() => {
+            const results = itemsMap.filter((item) => item.label.includes(filterQuery.toLowerCase()));
```

Comment:
You can store `filterQuery.toLowerCase()` in new variable so that you won't fire toLowerCase with each filter iteration